core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -94,6 +94,7 @@ public void accept(ErrorResponse error) {
           throw new CommitFailedException("Commit failed: %s", error.message());
         case 500:
         case 502:
+        case 503:
```

Comment:
Should we just change this to that any `5XX` code throws `CommitStateUnknownException`?

Reply:
I think the concern is that we're being conservative on what we consider erroneous and very aggressive about cleaning up in these cases.  We've been through this same issue multiple times now around 5XX codes and if we consider the commit state unknown for all 5XX codes, the only downside is that we leave more files around.  The side-effect of being too aggressive on cleanup is that we break a table, which is the worst option.